agents/agents-tools/src/commonMain/kotlin/ai/koog/agents/core/tools/Tool.kt

```diff
@@ -122,6 +138,30 @@ public abstract class Tool<TArgs, TResult> {
      */
     public fun encodeArgs(args: TArgs): JsonObject = ToolJson.encodeToJsonElement(actualArgsSerializer, args).jsonObject
 
+    /**
+     * Encodes the given result into a JSON representation using the configured result serializer.
+     *
+     * @param result The result object of type TResult to be encoded.
+     * @return A JsonObject representing the encoded result.
+     */
+    public fun encodeResult(result: TResult): JsonObject =
```

Comment:
Why is tool result an object? I can return primitive value such as `String` or `Int` from the `execute` method. Shouldn't it be `JsonElement` instead then?